account_index_based_currency/models/account_invoice.py

```diff
@@ -18,7 +18,7 @@ def _compute_currency_rate(self):
                 inv.company_id, inv.date_invoice or fields.Date.today())
 
     @api.depends('amount_total_signed', 'agreement_currency_rate',
-                 'date_invoice')
+                 'agreement_currency_amount', 'date_invoice')
```

Comment:
Shouldn't be both?
Since that you are using both to compute the values

Reply:
Both are used, agreement_currency_amount & agreement_currency_rate,

`agreement_currency_rate` is in the line above. It seems you have missed.

Regards